Yoast/wordpress-seo
# Don't output person stuff if no person is selected

## Summary

This PR can be summarized in the following changelog entry:

* When the site is set to represent a person, and no person is selected, don't output it in the schema.

## Technical decisions

* Decided to make it such that if no person has been selected yet, the website does not represent a person yet, but the context object has `false` for `site_represents`. This makes it far more reliable everywhere.

## Test instructions

This PR can be tested by following these steps:

* Make sure the site entity is set to `Person` and no user is selected.
	* Easy way (existing installation and with some developer magic ✨)
		1. Go to `SEO` > `Search Appearance` > `General`.
		2. Make sure `Choose whether the site represents an organisation or a person.` is set to `Person`.
		3. Run the following command in the console:
		`document.getElementById("person_id").value = null`
		4. Save the options.
	* Clean install
		1. Create a new WordPress install and install `Yoast SEO`.
		2. In the admin: Go to `SEO` > `Search Appearance` > `General`.
		3. Make sure `Choose whether the site represents an organisation or a person.` is set to `Person`.
		4. Don't select a user and save the options.
* View a post on the frontend.
* Run the page source through the [structured data testing tool](https://search.google.com/structured-data/testing-tool/u/1/?hl=en).
* Make sure there is no `publisher` attribute under `Article` > `isPartOf` > `isPartOf`.


## UI changes

## Quality assurance


Fixes #12812
